contracts/interstake-yield-generator/src/contract.rs

```diff
@@ -277,35 +279,77 @@ mod execute {
                 have: stake_details.total.amount,
             })?;
 
-        let msgs = delegate_msgs_for_validators(deps.as_ref(), amount.clone(), false)?;
-
-        TOTAL.update(deps.storage, |total| -> StdResult<_> {
-            Ok(coin((total.amount - amount.amount).u128(), total.denom))
-        })?;
+        STAKE_DETAILS.save(deps.storage, &info.sender, &stake_details)?;
 
+        // IMPORTANT: This will only queue the undelegation. The property processed is set to false.
         // Unbonding will result in coins going back to contract.
         // Create a claim to later be able to get tokens back.
-        let release_timestamp = env
-            .block
-            .time
-            .plus_seconds(config.unbonding_period.seconds());
         UNBONDING_CLAIMS.update(deps.storage, &info.sender, |vec_claims| -> StdResult<_> {
             let mut vec_claims = vec_claims.unwrap_or_default();
             vec_claims.push(ClaimDetails {
-                release_timestamp,
+                release_timestamp: None,
                 amount: amount.clone(),
             });
             Ok(vec_claims)
         })?;
 
-        STAKE_DETAILS.save(deps.storage, &info.sender, &stake_details)?;
-
         Ok(Response::new()
-            .add_attribute("action", "undelegate")
+            .add_attribute("action", "queue_undelegate")
             .add_attribute("sender", info.sender.to_string())
-            .add_attribute("amount", amount.to_string())
-            .add_attribute("release_timestamp", release_timestamp.to_string())
-            .add_messages(msgs))
+            .add_attribute("amount", amount.to_string()))
+    }
+
+    pub fn reconcile(
+        deps: DepsMut,
+        env: Env,
+        _info: MessageInfo,
+    ) -> Result<Response, ContractError> {
+        UNBOND_INFO.update(
+            deps.storage,
+            |info: UnbondInfo| -> Result<_, ContractError> { info.unbond_now(env.block.time) },
+        )?;
```

Comment:
So the main gimmick of this helper is - at instantiation you set buffer `current_time - 4 days`.
Then during undelegation you check if that time plus four days is smaller or equal then current time... Which means if at least four days has passed... If so, you set new time as current.

Reply:
Yea screw that, it was way to unclear. I've replaced the UnbondingInfo in its entirety